src/Datagrid/Datagrid.php

```diff
@@ -137,8 +137,9 @@ function ($value) {
 
         foreach ($this->getFilters() as $name => $filter) {
             $this->values[$name] = isset($this->values[$name]) ? $this->values[$name] : null;
-            if ($filter->isActive()) {
-                $filter->apply($this->query, $data[$filter->getFormName()]);
+            $filterFormName = $filter->getFormName();
```

Comment:
Why dont you use $name directly?

Reply:
It didn't work - Ironically if you use $name and you have a relation it will be something like:

$name has relations separated by periods
$filter->getFormName() has relations separated by __ (double underscore)
$this->values has the __ (double underscore) version of the key.
  (This could be a bug in the line above where $this->values[$name] is being set to null, but I didn't want to touch that code in case it breaks something else)

Example:
```
$name: comments.userId
$this->values: 
   [ 'comments__userId' => [ 'value' => "", ... ], 
    ...]
$filter->getFormName(): comments__userId
```